src/Cmdlets/GetKubeDeploymentCmdlet.cs

```diff
@@ -13,30 +13,47 @@ namespace Kubectl.Cmdlets {
     [Cmdlet(VerbsCommon.Get, "KubeDeployment")]
     [OutputType(new[] { typeof(DeploymentV1) })]
     public sealed class GetKubeDeploymentCmdlet : KubeApiCmdlet {
+        private const string DefaultParamSet = "DefaultParamSet";
+        private const string NamespaceObjectSet = "NamespaceObjectSet";
+
         [Parameter(ValueFromPipelineByPropertyName = true)]
         [Alias("Ns")]
         public string Namespace { get; set; } = "default";
 
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = NamespaceObjectSet,
+            ValueFromPipeline = true)]
+        [ValidateNotNull()]
+        public NamespaceV1 NamespaceObject { get; set; }
```

Comment:
Where we take object input, in addition to supporting the .NET class instances returned by `Get-` cmdlets, we should also support `PSObject`s as for example returned by `ConvertFrom-KubeYaml`. This could be accomplished with an `InputObject` parameter of type `object` and using `GetPropertyValue()`.

Reply:
If you have an example of this working, I'll take a crack at supporting this. I'm only familiar with how Azure PowerShell works internally and that's where I'm modeling my knowledge from.